tests/DCPS/UnionTopic/UnionTopic.cpp

```diff
@@ -0,0 +1,460 @@
+#include <dds/DdsDcpsInfrastructureC.h>
+#include <dds/DCPS/Service_Participant.h>
+#include <dds/DCPS/Marked_Default_Qos.h>
+#include <dds/DCPS/WaitSet.h>
+#include <dds/DCPS/SafetyProfileStreams.h>
+
+using OpenDDS::DCPS::retcode_to_string;
+
+#include "UnionTopicTypeSupportImpl.h"
+
+using namespace UnionTopic;
+
+#include <map>
+#include <string>
+
+const DDS::Duration_t max_wait_time = {10, 0};
+const int domain = 142;
+
+class Test {
+public:
+  DDS::DomainParticipant_var& participant_;
+  DDS::Publisher_var& publisher_;
+  DDS::Subscriber_var& subscriber_;
+  DDS::DataReader_var& reader_;
+  DDS::DataWriter_var& writer_;
+
+  Test(
+    DDS::DomainParticipant_var& participant,
+    DDS::Publisher_var& publisher,
+    DDS::Subscriber_var& subscriber,
+    DDS::DataReader_var& reader,
+    DDS::DataWriter_var& writer)
+    : participant_(participant)
+    , publisher_(publisher)
+    , subscriber_(subscriber)
+    , reader_(reader)
+    , writer_(writer)
+  {
+  }
+
+  bool
+  setup_test(DDS::Topic_var& topic)
+  {
+    DDS::ReturnCode_t rc;
+
+    // Create Subscriber
+    DDS::SubscriberQos subscriber_qos;
+    rc = participant_->get_default_subscriber_qos(subscriber_qos);
```

Comment:
Why not use SUBSCRIBER_QOS_DEFAULT?

Reply:
This test is based on Presentation, so that's an oversight on my part.